protocol/communities/persistence.go

```diff
@@ -1325,3 +1325,36 @@ func decodeEventsData(eventsBytes []byte, eventsDescriptionBytes []byte) (*Event
 		Events:                         events,
 	}, nil
 }
+
+func (p *Persistence) GetCommunityRequestsToJoinWithRevealedAddresses(communityID []byte) ([]*RequestToJoin, error) {
+	requests := []*RequestToJoin{}
+	rows, err := p.db.Query(`
+	SELECT id, public_key, clock, ens_name, chat_id, state
+	FROM communities_requests_to_join
+	WHERE req.community_id = ?`, communityID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	for rows.Next() {
+		request := &RequestToJoin{}
+		err = rows.Scan(&request.ID, &request.PublicKey, &request.Clock, &request.ENSName, &request.ChatID, &request.State)
+		if err != nil {
+			return nil, err
+		}
+
+		revealedAccounts, err := p.GetRequestToJoinRevealedAddresses(request.ID)
```

Comment:
Instead of doing two requests, you probably can do a JOIN in the above request

Reply:
not sure that it is possible, as we have a request to join + and an array of revealed account data
If there were only one row - agree.
P.S. Yes, here it is a bottleneck 😢

Reply:
I think it should be possible.

You do a query that returns all the revealed addresses and also the requests to join. Of course, the request to join data would be repeated for all the revealed addresses.

Then you just have to group per request ID (they should already be ordered actually.
You then loop them and when a new request to join arrives, you add to a new entry in the slice.

Does it make sense?